src/ParallelTestRunner.jl

```diff
@@ -212,58 +215,57 @@ end
 
 #
 # entry point
-#
-
-function runtest(::Type{TestRecord}, f, name, init_code, color)
-    function inner()
-        # generate a temporary module to execute the tests in
-        mod = @eval(Main, module $(gensym(name)) end)
-        @eval(mod, import ParallelTestRunner: Test, Random)
-        @eval(mod, using .Test, .Random)
-
-        Core.eval(mod, init_code)
+# 
 
-        data = @eval mod begin
-            GC.gc(true)
-            Random.seed!(1)
-
-            mktemp() do path, io
-                stats = redirect_stdio(stdout=io, stderr=io) do
-                    @timed try
-                        @testset $name begin
-                            $f
-                        end
-                    catch err
-                        isa(err, Test.TestSetException) || rethrow()
+function execute(::Type{TestRecord}, mod, f, name, color, custom_args)::TestRecord
+    data = @eval mod begin
+        GC.gc(true)
+        Random.seed!(1)
 
-                        # return the error to package it into a TestRecord
-                        err
+        mktemp() do path, io
+            stats = redirect_stdio(stdout=io, stderr=io) do
+                @timed try
+                    @testset $name begin
+                        $f
                     end
-                end
-                close(io)
-                output = read(path, String)
-                (; testset=stats.value, output, stats.time, stats.bytes, stats.gctime)
+                catch err
+                    isa(err, Test.TestSetException) || rethrow()
 
+                    # return the error to package it into a TestRecord
+                    err
+                end
             end
+            close(io)
+            output = read(path, String)
+            (; testset=stats.value, output, stats.time, stats.bytes, stats.gctime)
         end
+    end
 
-        # process results
-        rss = Sys.maxrss()
-        record = TestRecord(data..., rss)
+    # process results
+    rss = Sys.maxrss()
+    record = TestRecord(data..., rss)
 
-        GC.gc(true)
-        return record
-    end
+    GC.gc(true)
+    return record
+end
+
+function runtest(RecordType::Type{<:AbstractTestRecord}, f, name, init_code, color, custom_args)
+    # generate a temporary module to execute the tests in
+    mod = Core.eval(Main, Expr(:module, true, gensym(name), Expr(:block)))
```

Comment:
Now that this isn't in the custom_record_init anymore we can write it again as:

```suggestion
    mod = @eval(Main, module $(gensym(name)) end)
```